lib3h/src/mock_engine.rs

```diff
@@ -0,0 +1,134 @@
+use std::sync::mpsc;
+
+use crate::{network_engine::NetworkEngine, p2p_protocol::P2pProtocol};
+use holochain_lib3h_protocol::{protocol::Lib3hProtocol, Lib3hResult};
+
+/// Lib3h's 'mock mode' as a NetworkEngine
+pub struct MockEngine {
```

Comment:
I *think* we can do this as part of the real lib3h networking engine.

Even for real networking, if there are multiple agentIds in the same DNA in the same conductor, and they want to send messages / make requests of each other, we'll want to short-circuit that communication path and redirect it back to core.

Some things will be a little hacky / stub-y like until we have a dht / gossip module, we'll also need to reflect DHT fetch requests back as well. But as we iterate and add new modules, we can slowly add real functionality to it.

Thoughts? I just think it'd be cool to only have one real module to maintain going forward.

Reply:
Fine by me.